### PR TITLE
Add build workflow

### PR DESCRIPTION
**Pull request by crazy-max**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: build
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'master'
+      - 'v2'
```
Comment (tonistiigi) [inline .github/workflows/build.yml:8]: Don't need the v2 branch anymore

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
```
Comment (tonistiigi) [inline .github/workflows/build.yml:22]: Was there a reason the checkout action was needed here?

Comment (crazy-max) [reply .github/workflows/build.yml:20]: The bake action can't resolve git context like build-push action does atm. I think we could impl a remote context detection for some CI providers based on env vars like codecov does with their [bash uploader script](https://github.com/codecov/codecov-bash/blob/master/codecov) on buildx repo:

```hcl
target "base" {
    inherits = ["meta-helper"]
    providers = ["github", "travis"]
    context = "base"
    target = "base"
}
```

WDYT?

Comment (tonistiigi) [reply .github/workflows/build.yml:20]: Or it could just be the default if there is no local bake file and there is github actions context.

**Review by tonistiigi — APPROVED** [review]

**Comment by tonistiigi** [comment]
@crazy-max amd64 only 
image

**Comment by crazy-max** [comment]
@tonistiigi My bad, should be `base-all`

**Pull request merged**